swift/apple/FirezoneKit/Sources/FirezoneKit/Views/SessionView.swift

```diff
@@ -0,0 +1,132 @@
+//
+//  SessionView.swift
+//  (c) 2024 Firezone, Inc.
+//  LICENSE: Apache-2.0
+//
+
+import Combine
+import NetworkExtension
+import OSLog
+import SwiftUI
+
+@MainActor
+public final class SessionViewModel: ObservableObject {
+  @Published private(set) var actorName: String? = nil
+  @Published private(set) var resources: [Resource]? = nil
+  @Published private(set) var status: NEVPNStatus? = nil
+
+  let store: Store
+
+  private var cancellables: Set<AnyCancellable> = []
+
+  public init(store: Store) {
+    self.store = store
+
+    store.$actorName
+      .receive(on: DispatchQueue.main)
+      .sink(receiveValue: { [weak self] actorName in
+        guard let self = self else { return }
+
+        self.actorName = actorName
+      })
+      .store(in: &cancellables)
+
+    // MenuBar has its own observer
+    #if os(iOS)
+    store.$status
+      .receive(on: DispatchQueue.main)
+      .sink(receiveValue: { [weak self] status in
+        guard let self = self else { return }
+        self.status = status
+
+        if status == .connected {
+          store.beginUpdatingResources() { data in
+            self.resources = try? JSONDecoder().decode([Resource].self, from: data)
+          }
+        } else {
+          store.endUpdatingResources()
+        }
+
+      })
+      .store(in: &cancellables)
+    #endif
+  }
+
+  func signOutButtonTapped() {
+    Task {
+      try await store.signOut()
+    }
```

Comment:
Does `Task{}` launch a fire-and-forget green thread, like `go` in Golang?

Reply:
Exactly

Reply:
You can use it to encapsulate `async / await`